tests/e2e_tests/test_submit.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from playwright.sync_api import Page, expect
+
+from tests.e2e_tests.helpers import (
+    extract_email_link,
+    lookup_confirmation_emails,
+)
+from tests.e2e_tests.pages.find import DownloadDataPage
+from tests.e2e_tests.pages.submit import (
+    SubmitDashboardPage,
+    SubmitUploadPage,
+    SubmitUploadResponsePage,
+)
+
+pytestmark = pytest.mark.e2e
+
+
+def test_submit_report(domains, user_auth, page: Page):
+    PATH_TO_TEST_REPORTS = "tests/integration_tests/mock_pf_returns/"
+
+    dashboard_page = SubmitDashboardPage(page, domain=domains.submit)
+    dashboard_page.navigate()
+
+    expect(
+        page.get_by_role(
+            "heading",
+            name="Submit monitoring and evaluation data dashboard",
+        )
+    ).to_be_visible()
+
+    submit_upload_page: SubmitUploadPage = dashboard_page.select_fund("Pathfinders")
+
+    # test initial validation error upload
+    submit_upload_initial_error_page: SubmitUploadResponsePage = submit_upload_page.upload_report(
+        f"{PATH_TO_TEST_REPORTS}/PF_Round_1_Initial_Validation_Failures.xlsx"
+    )
+
+    expect(submit_upload_initial_error_page.get_title()).to_be_visible()
+    expect(submit_upload_initial_error_page.get_subtitle()).to_be_visible()
+
+    # test general validation error upload
+    submit_upload_general_error_page: SubmitUploadResponsePage = submit_upload_page.upload_report(
+        f"{PATH_TO_TEST_REPORTS}/PF_Round_1_General_Validation_Failures.xlsx"
+    )
+
+    expect(submit_upload_general_error_page.get_title()).to_be_visible()
+    expect(submit_upload_general_error_page.get_subtitle()).to_be_visible()
+
+    # test successful upload
+    submit_upload_success_page: SubmitUploadResponsePage = submit_upload_page.upload_report(
+        f"{PATH_TO_TEST_REPORTS}/PF_Round_1_Success.xlsx"
+    )
+
+    expect(submit_upload_success_page.get_title()).to_be_visible()
+    expect(submit_upload_success_page.get_subtitle()).to_be_visible()
+
+    la_email, fund_email = lookup_confirmation_emails(user_auth.email_address)
+
+    assert la_email is not None
+    assert fund_email is not None
+
+    fund_download_link = extract_email_link(fund_email)
+
+    download_page = DownloadDataPage(page)
```

Comment:
This one is a tricky one:
Retrieve download page and retrieve spreadsheet page are very similar and share same Playwright logic.

To avoid duplicating code, I created `DownloadPage` class and I created two empty classes `DownloadDataPage` and `RetrieveSpreadsheetPage` that extent `DownloadPage`. It might be over-engineering, but I thought this was the cleanest approach, if we want this Playwright POM approach.

I did something similar with the `SubmitUploadPage` (the template that has the upload file input). 
`SubmitUploadInitialErrorPage` and `SubmitUploadGeneralErrorPage` have the upload file input (so that users can retry upload), so they extend `SubmitUploadPage` to get access to the `upload_report` method.

If this is all too confusing and over-complicated, happy to have a quick chat about it to improve my approach. I think it's still fairly clean. But then again.. I feel about that about all code I ever write. :D

Reply:
I don't think it's really that tricky. They are separate pages and not related. They may look similar, but I don't think we should "de-duplicate" the code this prematurely. Just copy it. There is not some huge burden to have two separate POM playwright pages here that we're defeating with inheritance. It just feels overengineered :) 

Happy enough to leave the Submit pages as they are (as you've done them) though because yes those are a bit weird.